Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -264,7 +264,7 @@ void OpenSimContext::addPathPoint(GeometryPath& p, int menuChoice, PhysicalFrame
 
 bool OpenSimContext::deletePathPoint(GeometryPath& p, int menuChoice) {
     bool ret = p.deletePathPoint(*_configState, menuChoice );
-  _configState->invalidateAll(SimTK::Stage::Position);
+  _configState->invalidateAll(SimTK::Stage::Position); // This maynot be enough anymore in 4.0
```

Comment:
Deleting, adding, replacing, etc the `PathPoint` changes the model. The System should no longer be valid, in general. Look at your code for `replacePathPoint` that recreates the System after the edit.